apps/api/src/app/topics/use-cases/create-topic/create-topic.use-case.ts

```diff
@@ -1,17 +1,45 @@
-import { TopicEntity, TopicRepository } from '@novu/dal';
-import { ConflictException, Injectable } from '@nestjs/common';
+import {
+  CommunityOrganizationRepository,
+  EnvironmentEntity,
+  EnvironmentRepository,
+  OrganizationEntity,
+  TopicEntity,
+  TopicRepository,
+  UserEntity,
+} from '@novu/dal';
+import { ConflictException, Injectable, Logger, UnprocessableEntityException } from '@nestjs/common';
 
+import { FeatureFlagsKeysEnum, TOPIC_KEYS_REGEX } from '@novu/shared';
+import { FeatureFlagsService } from '@novu/application-generic';
 import { CreateTopicCommand } from './create-topic.command';
 
 import { TopicDto } from '../../dtos/topic.dto';
 
 @Injectable()
 export class CreateTopicUseCase {
-  constructor(private topicRepository: TopicRepository) {}
+  constructor(
+    private topicRepository: TopicRepository,
+    private featureFlagService: FeatureFlagsService,
+    private environmentRepository: EnvironmentRepository,
+    private communityOrganizationRepository: CommunityOrganizationRepository
+  ) {}
 
   async execute(command: CreateTopicCommand) {
     const entity = this.mapToEntity(command);
 
+    const [environment, organization] = await Promise.all([
+      this.environmentRepository.findOne({ _id: command.environmentId }),
+      this.communityOrganizationRepository.findOne({ _id: command.organizationId }),
+    ]);
+
+    if (!environment) {
+      throw new UnprocessableEntityException('Environment not found');
+    }
+
+    if (!organization) {
```

Comment:
Checking for organization should happen before the environment. It's higher up in the modeling hierarchy.